client.cpp

```diff
@@ -0,0 +1,86 @@
+#include <iostream>
+#include "libs/headers/bank.h"
+#include <fcntl.h>
+#include <sys/mman.h>
+#include <sys/stat.h>
+#include <unistd.h>
+#include <semaphore.h>
+#include <string>
+#include <vector>
+#include <algorithm>
+
+void BankAction(const std::string& cmd, Bank*& bnk)
+{
+	static const std::vector<std::string> commands = 
+	{ "get_max", "get_min", "get_ballance", 
+	  "transfer", 
+	  "freeze", "unfreeze", "isfrozen",
+	  "credit_from", "deposit_to",
+	  "credit", "deposit"};
+
+	auto it = std::find(commands.begin(), commands.end(), cmd);
+
+	if(it == commands.end())
+	{
+		std::cout << "Invalid operation\n";
+		return;
+	}
+	std::size_t index = it - commands.begin();
+
+	switch(index)
+	{
+		case 0:
+		{
+			std::size_t cellNum = 0;
+			std::cin >> cellNum;
+                        std::cout << "The max balance of cell " << cellNum << " is " << bnk->getMaxBalance(cellNum) << std::endl;
+		}
+		case 1:
+		{
+			std::size_t cellNum = 0;
+                        std::cin >> cellNum;
+                        std::cout << "The min balance of cell " << cellNum << " is " << bnk->getMinBalance(cellNum) << std::endl;
+		}
+		case 2:
+		{
+			std::size_t cellNum = 0;
+                        std::cin >> cellNum;
+                        std::cout << "The current balance of cell " << cellNum << " is " << bnk->getBalance(cellNum) << std::endl;
+		}
+	}
+
+}
+
+int main(int argc,char** argv)
+{
+	const char* semaphName = "/BankSem";
+
+	sem_t* semaph = sem_open (semaphName, O_RDWR);
+	int x = 0;
+	sem_getvalue(semaph, &x);
+	if(x == 0)
+	{
+		sem_close(semaph);
+		std::cout << "No free spaces to operate the bank currently.\nPlease try again Later.\n";
+		return 0;
```

Comment:
Why end the program?

Reply:
The work with semaphores is not implemented correctly.

Reply:
You must have either one semaphore with value 1 for the entire bank, or an array of semaphores the size of the bank cells.